pravega/pravega-operator
# issue-464:Added support for JvmOption customization

Signed-off-by: prabhaker24 <prabhaker.saxena@dell.com>

### Change log description
This pr will enable the user to modify both controller and sgmentstore JVMOptions as well as options during runtime

### Purpose of the change
Fixes #464 

### What the code does
The code looks for any change in the jvm option or option in either of controller or segmentstore and if there is a change i.e user has changed any of the jvmOptions or options or has added a new value, in that case, we update the configmap and restart each pod one by one.
I am doing validation as well and is done by the method **validateConfigMap()** which ensures user doesn't change any of the following values:-
1) controller.containerCount
2) controller.container.count
3) pravegaservice.containerCount
4) pravegaservice.container.count
5) bookkeeper.bkLedgerPath
6) bookkeeper.ledger.path
7) controller.retention.bucketCount
8) controller.retention.bucket.count
9) controller.watermarking.bucketCount
10) controller.watermarking.bucket.count
11) pravegaservice.dataLogImplementation
12) pravegaservice.dataLog.impl.name
13) pravegaservice.storageImplementation
14) pravegaservice.storage.impl.name
15) storageextra.storageNoOpMode
16) storageextra.noOp.mode.enable

### How to verify it
1) Deploy Pravega cluster and update segmentstore JVM Option we should see that segmentstore configmap should get updated and each pod should restart sequentially.
2)  Deploy Pravega cluster and update controller JVM Option we should see that controller configmap should get updated and each pod should restart sequentially.
3) Deploy Pravega cluster and update Option we should see that controller and segmentstore configmap should get updated and each of the controller and ss pod should restart sequentially.
4) Try changing any of the above 16 values that the user is not supposed to change and the operator should error saying that the parameter is not supposed to be changed.
